### PR TITLE
Add support for cancellations in RCP 3.0+. #4

### LINKED ISSUE #4: Add support for cancellations in RCP 3.0
I'm not using this plugin myself but I got a report from an RCP customer that cancellations weren't working in version 3.0. I gave him a bit of custom code that fixed it. I'll work up a PR shortly.

### PR DESCRIPTION
**Pull request by ashleyfae**
Closes #4

### CODE REVIEW

**Comment by mcfarlan** [comment]
Thanks so much Ashley for getting this shipped! We'll get this merged shortly. Much appreciated!

File: rcp-stripe-elements.php [hunk 1]
```diff
@@ -95,4 +95,51 @@ function rcp_elements_is_update_card_page() {
...
+ * Checks if the membership can be cancelled
+ *
+ * @param bool           $can_cancel    Whether or not the membership can be cancelled.
+ * @param int            $membership_id ID of the membership being checked.
+ * @param RCP_Membership $membership    Membership object.
+ *
+ * @return bool Whether or not the membership can be cancelled.
+ */
+function rcp_elements_can_cancel( $can_cancel, $membership_id, $membership ) {
+	if ( 'stripe_elements' != $membership->get_gateway() ) {
```
Comment (chrisvanpatten) [inline rcp-stripe-elements.php]: Would like to see this as a strict equality check (`!==`), and also a non-yoda condition

File: rcp-stripe-elements.php [hunk 2]
```diff
@@ -95,4 +95,51 @@ function rcp_elements_is_update_card_page() {
...
+ *
+ * @return true|WP_Error True if the cancellation was successful, WP_Error on failure.
+ */
+function rcp_elements_cancel( $success, $gateway, $gateway_subscription_id, $membership_id, $membership ) {
+	// If it was already cancelled, bail.
+	if ( $success ) {
+		return $success;
+	}
+
+	if ( 'stripe_elements' != $gateway ) {
```
Comment (chrisvanpatten) [inline rcp-stripe-elements.php]: Strict check, but also potentially combine to `if ( $success || $gateway !== 'stripe_elements' )`

**Review by chrisvanpatten — CHANGES_REQUESTED** [review]
Thanks @nosegraze! I caught two small things, but want to note that we in no way expect you to make those changes. We're happy to correct those on our end before we merge. Thank you SO much for submitting the change. To be honest it's nice to hear people are using this in production!

**Comment by ashleyfae** [comment]
No problem - updated. :)

**Review by chrisvanpatten — APPROVED** [review]

**Pull request merged**